### PR TITLE
Refactor allow_edits to depend on the notification state

### PR DESCRIPTION
**Pull request by slawosz**
https://regulatorydelivery.atlassian.net/browse/COSBETA-1433

A lot of `allow_edits` references in code were introducing chaos, and they were not required.

### CODE REVIEW

File: cosmetics-web/spec/helpers/responsible_persons/notifications_helper_spec.rb [hunk 1]
```diff
@@ -106,10 +107,10 @@
 
   describe "#notification_summary_product_rows" do
     subject(:summary_product_rows) do
-      helper.notification_summary_product_rows(notification, allow_edits:)
+      helper.notification_summary_product_rows(notification)
     end
 
-    let(:allow_edits) { false }
+    let(:editable) { false }
```
Comment (scruti) [inline cosmetics-web/spec/helpers/responsible_persons/notifications_helper_spec.rb:113]: This doesn't seem used anywhere? :detective: 
```suggestion
```

File: cosmetics-web/spec/helpers/responsible_persons/notifications_helper_spec.rb [hunk 2]
```diff
@@ -239,11 +240,11 @@
 
   describe "#notification_summary_component_rows" do
     subject(:summary_component_rows) do
-      helper.notification_summary_component_rows(component, include_shades:, allow_edits:)
+      helper.notification_summary_component_rows(component, include_shades:)
     end
 
     let(:include_shades) { false }
-    let(:allow_edits) { false }
+    let(:editable) { false }
```
Comment (scruti) [inline cosmetics-web/spec/helpers/responsible_persons/notifications_helper_spec.rb:247]: This doesn't seem to be used anywhere :detective: 
```suggestion
```

File: cosmetics-web/spec/models/notification_spec.rb [hunk 3]
```diff
@@ -565,4 +565,12 @@
       end
     end
   end
+
+  describe "#editable?" do
+    it "is true for EDITABLE_STATES" do
+      Notification::EDITABLE_STATES.each do |state|
+        expect(build(:notification, state:)).to be_editable
+      end
+    end
```
Comment (scruti) [inline cosmetics-web/spec/models/notification_spec.rb:574]:
```suggestion
    it "is true for EDITABLE_STATES" do
      Notification::EDITABLE_STATES.each do |state|
        expect(build(:notification, state:)).to be_editable
      end
    end
    
    it "is false for complete/deleted states" do
      [Notification::NOTIFICATION_COMPLETE, Notification::DELETED].each do |state|
        expect(build(:notification, state:)).not_to be_editable
      end
    end
```

**Review by scruti — APPROVED** [review]

**Review by nasirkhanbeis — APPROVED** [review]
Regression tested this on the review app, edit draft is looking good as is on master.

**Pull request merged**